app/controllers/locationOfGoods/IdentificationController.scala

```diff
@@ -39,38 +42,58 @@ class IdentificationController @Inject() (
   implicit val sessionRepository: SessionRepository,
   navigatorProvider: LocationOfGoodsNavigatorProvider,
   actions: Actions,
+  getMandatoryPage: SpecificDataRequiredActionProvider,
   formProvider: EnumerableFormProvider,
+  locationOfGoodsIdentificationTypeService: LocationOfGoodsIdentificationTypeService,
   val controllerComponents: MessagesControllerComponents,
   view: IdentificationView
 )(implicit ec: ExecutionContext, phaseConfig: PhaseConfig)
     extends FrontendBaseController
     with I18nSupport {
 
-  private val form = formProvider[LocationOfGoodsIdentification]("locationOfGoods.identification")
+  private def form(locationOfGoodsIdentification: Seq[LocationOfGoodsIdentification]): Form[LocationOfGoodsIdentification] =
+    formProvider("locationOfGoods.locationOfGoodsIdentificationType", locationOfGoodsIdentification)
 
-  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn).async {
-    implicit request =>
-      LocationOfGoodsIdentification.values(request.userAnswers) match {
-        case identifier :: Nil =>
-          redirect(mode, InferredIdentificationPage, identifier)
-        case identifiers =>
-          val preparedForm = request.userAnswers.get(IdentificationPage) match {
-            case None        => form
-            case Some(value) => form.fill(value)
-          }
-
-          Future.successful(Ok(view(preparedForm, lrn, identifiers, mode)))
-      }
-  }
+  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions
+    .requireData(lrn)
+    .andThen(getMandatoryPage(LocationTypePage))
+    .async {
+      implicit request =>
+        val locationType = request.arg
+        locationType match {
+          case LocationType("B", _) =>
+            IdentificationPage.writeToUserAnswers(LocationOfGoodsIdentification("Y", "Authorisation number"))
+            Future.successful(Redirect(controllers.locationOfGoods.routes.AuthorisationNumberController.onPageLoad(lrn, mode)))
+          case _ =>
+            locationOfGoodsIdentificationTypeService.getLocationOfGoodsIdentificationTypes(request.userAnswers).map {
+              locationOfGoodsIdentificationTypes =>
+                val preparedForm = request.userAnswers.get(IdentificationPage) match {
+                  case None        => form(locationOfGoodsIdentificationTypes)
+                  case Some(value) => form(locationOfGoodsIdentificationTypes).fill(value)
+                }
+                Ok(view(preparedForm, lrn, locationOfGoodsIdentificationTypes, mode))
+            }
+        }
+    }
```

Comment:
As stated before, cannot this not just be...

```
Hmm not sure about this... We've removed quite an important bit of functionality here that stores a value inside the 'inferred' value path. The actual change needed frrom the original is quite minimal no? Something like:

```
def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn).async {
    implicit request =>
locationOfGoodsIdentificationTypeService.getLocationOfGoodsIdentificationTypes(request.userAnswers).map {
        case identifier :: Nil =>
          redirect(mode, InferredIdentificationPage, identifier)
        case identifiers =>
          val preparedForm = request.userAnswers.get(IdentificationPage) match {
            case None        => form
            case Some(value) => form.fill(value)
          }

          Future.successful(Ok(view(preparedForm, lrn, identifiers, mode)))
      }
}
```
```